File: Cpp/1572_Matrix_Diagonal_Sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int result = 0;
        int n = mat.size();
        for(int i=0; i<n; ++i){
            result += mat[i][i];
            result += mat[i][n-1-i];
        }
        
        if(n & 1){
            result -= mat[n/2][n/2];
        }
        
        return result;
    }
};

File: Kotlin/1572_Matrix_Diagonal_Sum.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        var result = 0
        var n = mat.size
        
        for(i in mat.indices){
            result += mat[i][i]
            result += mat[i][n - 1 - i]
        }
        
        if(n and 1 == 1){
            result -= mat[n / 2][n / 2]
        }
        
        return result
    }
}